### PR TITLE
Add ban-ts-ignore rule.

### PR DESCRIPTION
**Pull request by bowenni**
#### PR checklist

- [x] Addresses an existing issue: #3626
- [x] New feature, bugfix, or enhancement
  - [x] Includes tests
- [x] Documentation update

#### Overview of change:
Add a new rule to ban "// @ts-ignore" comments.

#### CHANGELOG.md entry:

[new-rule] `ban-ts-ignore`

### CODE REVIEW

**Comment by bowenni** [comment]
The failing test seems unrelated to my change?

**Comment by bowenni** [comment]
Friendly ping :)

**Comment by johnwiseheart** [comment]
Hi @bowenni - sorry for taking so long to review this PR. Theres a couple of new maintainers coming on to tslint - we're just ramping up to the codebase, then planning to fix up the failing test suite, and finally work through these PRs one by one and get them merged in. Thank you for your patience!

Until that time, a friendly reminder that you're able to use this rule locally as a custom rule in your `tsconfig.json` file.

File: src/rules/banTsIgnoreRule.ts [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+        description: 'Bans "// @ts-ignore" comments from being used.',
+        optionsDescription: "Not configurable.",
+        options: null,
+        optionExamples: [true],
+        type: "typescript",
+        typescriptOnly: true,
+    };
+    /* tslint:disable:object-literal-sort-keys */
+
+    public static FAILURE_STRING = 'Do not use "// @ts-ignore" comments because it suppresses compilation errors.';
```
Comment (no23reason) [inline src/rules/banTsIgnoreRule.ts]: Grammar nit: Do not use "// @ts-ignore" comments because *they suppress* compilation errors.

**Comment by aervin** [comment]
@bowenni The ability to add code examples to rule documentation just landed in master, would you mind adding some for this rule? The best example of how to do this (on master) is the `curly` rule.

**Comment by bowenni** [comment]
Sure @aervin .
How do I generate the docs locally?

**Comment by aervin** [comment]
Thanks @bowenni . We did some research on building docs locally [in this thread](https://github.com/palantir/tslint/pull/3602#issuecomment-357519433). The picture of the result may be enough for you to go on without having to actually build the docs.

**Comment by bowenni** [comment]
Thanks for the pointer @aervin .
I added a code example
oyypox20seh

File: src/rules/banTsIgnoreRule.ts [hunk 2]
```diff
@@ -0,0 +1,52 @@
+/**
+ * @license
+ * Copyright 2013 Palantir Technologies, Inc.
```
Comment (aervin) [inline src/rules/banTsIgnoreRule.ts]: Will you change the dates to 2018 in this file and the examples file?

Comment (bowenni) [reply src/rules/banTsIgnoreRule.ts]: Done.

File: src/rules/code-examples/banTsIgnore.examples.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+// tslint:disable: object-literal-sort-keys
+export const codeExamples = [
+    {
+        description: 'Disallows the use of "@ts-ignore"',
+        config: Lint.Utils.dedent`
+            "rules": { "ban-ts-ignore": true }
+        `,
+        pass: Lint.Utils.dedent`
+            console.log("hello");
```
Comment (aervin) [inline src/rules/code-examples/banTsIgnore.examples.ts]: I think use the same code snippet from the `fail` example, but change the comment text to
```ts
// Compiler warns about unreachable code error
```

Comment (bowenni) [reply src/rules/code-examples/banTsIgnore.examples.ts]: Done. Thanks!

File: src/rules/code-examples/curly.examples.ts [hunk 4]
```diff
@@ -1,6 +1,6 @@
 /**
  * @license
- * Copyright 2013 Palantir Technologies, Inc.
```
Comment (giladgray) [inline src/rules/code-examples/curly.examples.ts]: this change is not relevant. do not update copyright statements.

**Comment by bowenni** [comment]
Friendly ping :)

**Comment by johnwiseheart** [comment]
Hi @bowenni, can you please merge with master to ensure that the tests pass and then I can merge this.

**Comment by bowenni** [comment]
Thanks @johnwiseheart. Done.

**Comment by johnwiseheart** [comment]
Thanks for contributing!

**Comment by mickaeltr** [comment]
Hello,
That's maybe a dumb question by I am wondering why I can't see this rule in [the list](https://palantir.github.io/tslint/rules/)?
My project is using TSLint 5.11.0
Thanks

**Pull request merged**